### PR TITLE
Implement rename filesystem

### PR DESCRIPTION
**Pull request by trgill**
Added code for dbus and sim_engine.
Implemented rename filesystem for sim_engine
Update uuid to version 0.3.1 and enable v4 features (minor changes)
Make uuid key for filesystems table in the Pool
The uuid version change required minor updates to strat_engine

Signed-off-by: Todd Gill <tgill@redhat.com>

### CODE REVIEW

**Comment by trgill** [comment]
@agrover @mulkieran review please?

**Comment by mulkieran** [comment]
It's a pretty significant question. If the name is how the pool identifies filesystems, then everything, dbus and engine, should be a pool method. If the name is just an attribute of the filesystems, then everything should be a filesystem method, I think.

**Comment by mulkieran** [comment]
I just thought of this one while skating...the dbus map between object path and name will have to be updated somehow. Also the filesystem maps, because they include the pool name. I _think_ they should all be updated w/out changing the object path values.

**Comment by trgill** [comment]
I fixed the object path map issue - thanks.

I had been thinking we'd want attributes of our objects to be dbus "Properties".  But if that is the case, we don't need Rename at all.  It would just be a Set() call on a property.

Talking tomorrow probably makes sense.

File: src/dbus_api.rs [hunk 1]
```diff
@@ -355,7 +372,48 @@ fn set_filesystem_mountpoint(m: &MethodInfo<MTFn<TData>, TData>) -> MethodResult
...
+
+    let new_name: &str = try!(get_next_arg(&mut iter, 0));
+
+    let dbus_context = m.path.get_data();
+    let object_path = m.path.get_name();
+    let return_message = message.method_return();
+    let return_sig = "qs";
+    let default_return = MessageItem::Array(vec![], return_sig.into());
+
+    let pair = match object_path_to_pair(dbus_context, object_path) {
```
Comment (mulkieran) [inline src/dbus_api.rs]: You can use destructuring let and ```dbus_try!``` here like:

```let (pool_name, _) = dbus_try!(object_path_to_pair(dbus_context, object_path); default_return; return_message); ```

I think.

**Comment by trgill** [comment]
Updated to use UUID as a key for the filesystem table in Pool.

@mulkieran @agrover review please?

File: Cargo.toml [hunk 2]
```diff
@@ -24,3 +23,7 @@ term = "0.4"
 rand = "0.3"
 bidir-map = "0.3"
 serde = "0"
+
+[dependencies.uuid]
+version = "0.3.1"
+features = ["v4"]
```
Comment (agrover) [inline Cargo.toml]: EOL here please :smile:

Comment (trgill) [inline Cargo.toml]: Ok.  Thanks.

File: src/dbus_api.rs [hunk 3]
```diff
@@ -355,7 +372,60 @@ fn set_filesystem_mountpoint(m: &MethodInfo<MTFn<TData>, TData>) -> MethodResult
...
+    let default_return = MessageItem::Array(vec![], return_sig.into());
+
+    let (pool_name, filesystem_name) =
+        dbus_try!(object_path_to_pair(dbus_context, object_path); default_return; return_message);
+
+    let mut b_engine = dbus_context.engine.borrow_mut();
+    let ref mut pool = engine_try!(b_engine.get_pool(&pool_name);
+                                   default_return;
+                                   return_message);
+
```
Comment (trgill) [inline src/dbus_api.rs]: Added check to make sure new_name doesn't already exist.  Will return error now if new_name is found.

File: src/sim_engine/pool.rs [hunk 4]
```diff
@@ -99,6 +113,28 @@ impl Pool for SimPool {
...
+            Some(filesystem) => filesystem,
+            None => return Err(EngineError::Stratis(ErrorEnum::NotFound(id.simple().to_string()))),
+        };
+
+        Ok(return_filesystem)
+    }
+
+    fn get_filesystem_id(&mut self, name: &str) -> EngineResult<Uuid> {
+
+        for (_, value) in self.filesystems.iter().enumerate() {
```
Comment (mulkieran) [inline src/sim_engine/pool.rs]: You don't need enumerate here, soemthing like:

```
for (_, value) in self.filesystems.iter() {
    if value.has_same(name) {
        return Ok(value.get_id());
...
```

File: src/dbus_api.rs [hunk 5]
```diff
@@ -379,7 +396,67 @@ fn set_filesystem_mountpoint(m: &MethodInfo<MTFn<TData>, TData>) -> MethodResult
...
+            let (rc, rs) = engine_to_dbus_err(&error);
+            let (rc, rs) = code_to_message_items(rc, rs);
+            return Ok(vec![return_message.append2(rc, rs)]);
+        }
+        Err(_) => {}
+    }
+
+    let filesystem = match pool.get_filesystem_id(&filesystem_name) {
+        Ok(id) => {
+            match pool.get_filesystem(&id) {
```
Comment (mulkieran) [inline src/dbus_api.rs]: Might as well use ```engine_try_0!``` here.

Comment (trgill) [inline src/dbus_api.rs]: Ok - done

File: src/sim_engine/pool.rs (hunk 4 above)
Comment (trgill) [inline src/sim_engine/pool.rs]: Ok - done

**Comment by mulkieran** [comment]
:+1:  OK!

**Comment by agrover** [comment]
Merging, has approvals and it's blocking me :smile:

**Pull request merged**